### PR TITLE
Added Workshop access mixin and fixed workshop update form bug

### PR DESCRIPTION
**Pull request by SudShekhar**
Also, added database query for workshop_completed list.

### CODE REVIEW

File: wye/workshops/views.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def get_context_data(self, *args, **kwargs):
         context['workshop_list'] = workshop_list
         context['workshop_feedback_pending'] = []
         context['workshop_in_queue'] = []
-        context['workshop_completed'] = []
+        context['workshop_completed'] = Workshop.objects.filter(status=WorkshopStatus.COMPLETED)
```
Comment (vnbang2003) [inline wye/workshops/views.py]: Can you please add below query Workshop.objects.filter(status=WorkshopStatus.ACCEPTED)

File: wye/workshops/views.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ def get_context_data(self, *args, **kwargs):
...
         context['workshop_withdrawn'] = []
```
Comment (vnbang2003) [inline wye/workshops/views.py]: Workshop.objects.filter(active=False)

**Comment by vnbang2003** [comment]
Looks good. Let extend the query for each variables.

**Comment by vnbang2003** [comment]
I will merge this for now to keep moving.

File: wye/workshops/views.py (hunk 2 above)
Comment (SudShekhar) [inline wye/workshops/views.py]: ^ is that the logic for withdrawn workshops?

File: wye/workshops/views.py (hunk 1 above)
Comment (SudShekhar) [inline wye/workshops/views.py]: Where do you want this query? For the in_queue workshops?

File: wye/workshops/mixins.py [hunk 3]
```diff
@@ -1,5 +1,19 @@
...
+
+
+class WorkshopAccessMixin(object):
+
+    def dispatch(self, request, *args, **kwargs):
+        pk = self.kwargs.get(self.pk_url_kwarg, None)
+        workshop = Workshop.objects.get(id=pk)
+        if workshop.requester not in self.request.user.organisation_users.all():
+            print(workshop.requester)
+            print(self.request.user.organisation_users.all())
```
Comment (shankisg) [inline wye/workshops/mixins.py]: I see multiple print function in the mixin, we need to be careful about this.

Comment (SudShekhar) [inline wye/workshops/mixins.py]: I had planned to remove it with my next commit, had forgotten to remove the debug messages last time.

File: wye/workshops/views.py (hunk 2 above)
Comment (SudShekhar) [inline wye/workshops/views.py]: @vnbang2003 Waiting for further details regarding your comment. Will update create a fresh PR with additional queries and removed comments then.

**Pull request merged**